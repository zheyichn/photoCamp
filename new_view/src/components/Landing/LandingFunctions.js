import React from "react";
import {
  Grid,
  Box,
  TextField,
  Typography,
  Button,
  Link,
  Alert,
} from "@mui/material";
import { useTheme } from "@mui/material/styles";
import { ThemeProvider } from "@emotion/react";
import {
  loginFunctionBoxStyles,
  createAccountBoxStyles,
} from "../../styles/landingStyle";
import { login } from "../../api/mock_api_j";
import { useState } from "react";
// import { styled } from "@mui/material/styles";

export default function LandingFunctions() {
  const theme = useTheme();

  // capture the current state of log in process
  const [success, setSuccess] = useState(false);
  const [failure, setFailure] = useState(false);
  const [user, setUser] = useState("");
  const [errorMsg, setErrorMsg] = useState("");

  // process login function
  const handleLogin = async () => {
    setFailure(false);
    const res = await login(userName, password);
    if (!res) {
      setErrorMsg("Connection not found");
      setFailure(true);
      return;
    }
    if (res.status !== 201) {
      setErrorMsg(res.data.message);
      setFailure(true);
      return;
    }
    setUser(res.data.userName);
    setSuccess(true);
    setTimeout(() => {
      setSuccess(false);
      setUser("");
      window.location.pathname = "/homepage";
    }, 2000);
  };

  const handleNewAccountClick = () =>
    (window.location.pathname = "/createnewaccount");

  // handles user inputs
  let userName;
  let password;

  const handleOnChange = (e) => {
    if (e.target.name === "Username:") {
      userName = e.target.value;
    }
    if (e.target.name === "Password:") {
      password = e.target.value;
    }
  };

  // Text boxes used
  const LogInTextField = (props) => {
    const myProp = props;
    const label = myProp.label;
    return (
      <TextField
        id={props.id}
        label={label}
        name={label}
        variant="outlined"
        size="small"
        autoComplete="off"
        onChange={handleOnChange}
      />
    );
  };

  return (
    <ThemeProvider theme={theme}>
      <Grid
        container
        spacing={0}
        direction="row"
        alignItems="center"
        justifyContent="center"
        sx={{ width: "500px", height: "150px" }}
      >
        <Grid item xs={6} sx={{ borderRight: 1, height: "200px" }}>
          <Box m={2} sx={loginFunctionBoxStyles}>
            <LogInTextField label="Username:" id="userNameInput" />
            <LogInTextField label="Password:" id="passwordInput" />
            {failure && <Alert severity="error">{errorMsg}</Alert>}
            {success && (
              <Alert severity="success">Welcome {user}. Please wait...</Alert>
            )}
            <Button
              onClick={handleLogin}
              variant="contained"
              size="small"
              sx={{ width: "80px" }}
              id="signinBtn"
            >
              Sign In
            </Button>
            <Typography style={{ fontSize: 12 }}>
              <Link href="/forgotpassword">Forgot Password?</Link>
            </Typography>
          </Box>
        </Grid>
        <Grid item xs={6} style={{ height: "150px" }}>
          <Box sx={createAccountBoxStyles}>
            <Button
              onClick={handleNewAccountClick}
              variant="contained"
              size="small"
              sx={{ width: "140px" }}
              id="registerBtn"
            >
              <Link href="/CreateNewAccount" underline="none" color="white">
                Create
                <br />
                New Account
              </Link>
            </Button>
          </Box>
        </Grid>
      </Grid>
    </ThemeProvider>
  );
}
