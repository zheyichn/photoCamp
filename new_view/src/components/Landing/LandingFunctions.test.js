import { render, screen, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import LandingFunctions from "./LandingFunctions";

// afterEach function runs after each test suite is executed
afterEach(() => {
    cleanup();
});

describe("Landing Functions", () => {
  
    // Test1: match snapshots
    it("Page renders correctly", () => {
      const tree = renderer
        .create(
          <LandingFunctions />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    // Test2: check log in text fields
    it("Log in textboxes handle input correctly", async() => {
      render(<LandingFunctions />);
      const element = screen.getAllByRole("textbox");
      await userEvent.type(element[0], "this is a username");
      await userEvent.type(element[1], "this is a password");

      // assertion: verify that the text is in the textbox
      expect(element[0]).toHaveValue("this is a username");
      expect(element[1]).toHaveValue("this is a password");
    })

});