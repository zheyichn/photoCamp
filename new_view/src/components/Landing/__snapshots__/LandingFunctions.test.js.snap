// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Functions Page renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-6qlfzj-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-dlw3eo-MuiGrid-root"
  >
    <div
      className="MuiBox-root css-1pdo0d4"
    >
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r0:"
          id=":r0:-label"
        >
          Username:
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r0:"
            name="Username:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Username:
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor=":r1:"
          id=":r1:-label"
        >
          Password:
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id=":r1:"
            name="Password:"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Password:
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-dw74n3-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Sign In
      </button>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        style={
          {
            "fontSize": 12,
          }
        }
      >
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
          href="/forgotpassword"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Forgot Password?
        </a>
      </p>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    style={
      {
        "height": "150px",
      }
    }
  >
    <div
      className="MuiBox-root css-1vcrsgd"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-h1cs6j-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-rmjmia-MuiTypography-root-MuiLink-root"
          href="/CreateNewAccount"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Create
          <br />
          New Account
        </a>
      </button>
    </div>
  </div>
</div>
`;
